earth_enterprise/src/fusion/autoingest/sysman/khResourceManager.cpp

```diff
@@ -925,20 +925,26 @@ void
 khResourceManager::GetCurrTasks(TaskLists &ret)
 {
   assert(!mutex.TryLock());
-  for (TaskWaitingQueue::iterator t = taskWaitingQueue.begin();
-       t != taskWaitingQueue.end(); ++t) {
-    khTask *task = *t;
-    ret.waitingTasks.push_back
-      (TaskLists::WaitingTask(task->verref(),
-                              task->taskid(),
-                              task->priority(),
-                              task->submitTime(),
-                              task->activationError));
-  }
-
-  for (Providers::iterator p = providers.begin();
-       p != providers.end(); ++p) {
-    p->second->AddToTaskLists(ret);
+  try {
+    for (TaskWaitingQueue::iterator t = taskWaitingQueue.begin();
+        t != taskWaitingQueue.end(); ++t) {
```

Comment:
Could this be replaced with a range loop?

Reply:
Turns out we can't without changing the data structure (it's `std::set` right now). But at this point we're planning to keep the lock around this section of code, so we won't have to worry about mulit-threaded access.